src/main/java/net/minecraftforge/common/EnumPlantType.java

```diff
@@ -8,5 +10,27 @@
     Cave,
     Water,
     Nether,
-    Crop
-}
\ No newline at end of file
+    Crop;
+
+    /**
+     * Getting a custom {@link EnumPlantType}, or an existing one if it has the same name as that one. Your plant should implement {@link IPlantable}
+     * and return this custom type in {@link IPlantable#getPlantType(net.minecraft.world.IBlockAccess, net.minecraft.util.BlockPos)}.
+     * 
+     * <p>If your new plant grows on blocks like any one of them above, never create a new {@link EnumPlantType}! 
+     * This enumeration is only functioning in 
+     * {@link net.minecraft.block.Block#canSustainPlant(net.minecraft.world.IBlockAccess, net.minecraft.util.BlockPos, net.minecraft.util.EnumFacing, IPlantable)},
+     * which you are supposed to override this function in your new block and create a new plant type to grow on that block!
+     * 
+     * <p>For compatibility you should use this method as often as you can for new types, instead of keeping an instance and let others access it.
+     * @param name the name of the type of plant, you had better follow the style above
```

Comment:
Actually no, caching it once is the best practice as a GETFIELD is MUST faster then the loop below. And will result in the same value.